input.lisp

```diff
@@ -37,9 +38,88 @@
           read-one-char
           read-one-line))
 
+
+;;; General Utilities
+
+(defun take (n list)
+  "Returns a list with the first n elements of the given list."
+  (loop repeat n
+        for x in list
+        collect x))
+
+;; This could use a much more efficient algorithm.
+;; But for our purposes with small lists it's likely ok.
+(defun longest-common-prefix (seqs &key (test #'eql))
```

Comment:
https://github.com/joaotavora/sly/blob/c338f22cc2b88c05b80b417b8a7a13b3e3346ca9/slynk/slynk-completion.lisp#L70

We could re-use this implementation. In particular because it uses reduce instead of apply which can reach the maximum argument limit.